CDP4Composition/ViewModels/DockLayoutViewModel.cs

```diff
@@ -0,0 +1,119 @@
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft, Nathanael Smiechowski, Ahmed Abulwafa Ahmed
+//
+//    This file is part of CDP4-IME Community Edition.
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Composition.ViewModels
+{
+    using System.ComponentModel.Composition;
+    using System.Linq;
+    using System.Reactive;
+    using System.Threading.Tasks;
+
+    using CDP4Composition.Navigation;
+
+    using DevExpress.Xpf.Docking;
+    using DevExpress.Xpf.Docking.Base;
+
+    using ReactiveUI;
+
+    /// <summary>
+    /// This is the view model for the main application docking control. 
+    /// </summary>
+    [Export]
+    public class DockLayoutViewModel : ReactiveObject
+    {
+        private readonly IDialogNavigationService dialogNavigationService;
+
+        [ImportingConstructor]
+        public DockLayoutViewModel(IDialogNavigationService dialogNavigationService)
+        {
+            DockPanelViewModels = new ReactiveList<IPanelViewModel>();
+
+            this.dialogNavigationService = dialogNavigationService;
+
+            DockPanelClosingCommand = ReactiveCommand.CreateAsyncTask(arg => PanelClosing((ItemCancelEventArgs)arg));
+            DockPanelClosedCommand = ReactiveCommand.CreateAsyncTask(arg => PanelClosed((DockItemClosedEventArgs)arg));
+        }
+
+        /// <summary>
+        /// The panel items of the dock
+        /// </summary>
+        public ReactiveList<IPanelViewModel> DockPanelViewModels { get; }
+
+        /// <summary>
+        /// Gets the command that is called when the dock panel is closing
+        /// </summary>
+        public ReactiveCommand<Unit> DockPanelClosingCommand { get; }
+
+        /// <summary>
+        /// Gets the command that is called when the dock panel is closed
+        /// </summary>
+        public ReactiveCommand<Unit> DockPanelClosedCommand { get; }
+
+        private async Task PanelClosed(DockItemClosedEventArgs e)
```

Comment:
I think this is a quirk of the current version of Reactive we're using. This seems to be the odd convention adopted in the rest of the solution.

Reply:
The command I need:
https://www.reactiveui.net/api/reactiveui/reactivecommand/303b463f
doesn't appear to be available in the version of reactive that's currently being used in IME.

So I have to satisfy the async interface in order to use the command. However as the async method does not contain anything asynchronous it will effectively run synchronously which is the behaviour I want.